bin/generate_inputs.py

```diff
@@ -0,0 +1,221 @@
+    parser.add_argument('-r', '--raw-reads-study', required=True,
+                        help='Raw reads study accession')
+    parser.add_argument('-o', '--outdir', required=False, default='outdir',
+                        help='Output directory')
+    parser.add_argument('--output-samplesheet', required=False,
+                        help='Name of output file', default="samplesheet.csv")
+    parser.add_argument('-b', '--scientific-name', required=False,
+                        help='Comma separated list of scientific_name(s) that would be included')
+    parser.add_argument('-e', '--environment-biome', required=False,
+                        help='Comma separated list of environment_biome(s) that would be included')
+    parser.add_argument('--keep-metat', action='store_true')
+    return parser.parse_args()
+
+
+def transform_paths(ftp_path: str) -> Tuple[str, str]:
+    """
+    Transforms an FTP path according to privacy, it also returns if it's public or private. There are some limitations.
+    Use https:// for public (-resume nextflow works for https:// but not for ftp and s3)
+    Use FIRE S3 object key for private data (-resume is broken for S3 paths)
+    """
+    if ftp_path.startswith("ftp.sra.ebi.ac.uk/vol1/"):
+        #s3_key = ftp_path.replace("ftp.sra.ebi.ac.uk/vol1/", "s3://era-public/")
+        https_key = 'https://' + ftp_path
+        print(f"Detected a public file for FTP path: {ftp_path}")
+        return https_key, 'public'
+    elif ftp_path.startswith("ftp.dcc-private.ebi.ac.uk/vol1/"):
+        s3_key = ftp_path.replace("ftp.dcc-private.ebi.ac.uk/vol1/", "s3://era-private/")
+        print(f"Detected a private file for FTP path: {ftp_path}")
+        return s3_key, 'private'
+    else:
+        raise ValueError(
+            f"Invalid FTP path: {ftp_path}. Must start with 'ftp.sra.ebi.ac.uk/vol1/' or 'ftp.dcc-private.ebi.ac.uk/vol1/'."
+        )
+
+
+def load_xml(assembly):
+    retry_attempts = 3
+    retry_delay_min = 15
+    xml_url = "https://www.ebi.ac.uk/ena/browser/api/xml/{}".format(assembly)
+
+    for attempt in range(1, retry_attempts + 1):
+        r = requests.get(xml_url)
+        if not r.ok:
+            retry_delay = retry_delay_min * attempt
+            if r.status_code == 500:
+                print(f"Received HTTP 500 error for sample {assembly}. Retrying in {retry_delay} seconds...")
+                time.sleep(retry_delay)
+                continue
+            else:
+                print(f"Unable to fetch xml for sample {assembly}. Retrying in {retry_delay} seconds...")
+                time.sleep(retry_delay)
+                continue
+        if r.ok:
+            data_dict = xmltodict.parse(r.content)
+            json_dump = json.dumps(data_dict)
+            json_data = json.loads(json_dump)
+            return json_data
+        else:
+            print("Could not retrieve xml for accession", assembly)
+            print(r.text)
+            return None
+
```

Comment:
This is repeated in the genomes-catalogue pipeline.. not for this PR, but we should incorporate this into the ena api (probably if/when we extract that from emgapi v2).